module_utils/foreman_helper.py

```diff
@@ -309,6 +311,7 @@ def __init__(self, argument_spec, **kwargs):
 
 
 def sanitize_entity_dict(entity_dict, name_map):
+    name_map['id'] = 'id'
```

Comment:
I think, this is introducing a side effect: `name_map` will be modified after a call to this function. Is this a serious problem?

Reply:
I don't think so. I just want `id → id` to always be in the map.

Reply:
(and `id` should never be mapped to anything else, riiiiiight?)

Reply:
I thought this would be nice:
```
actual_map = collections.ChainMap(name_map, DEFAULT_MAPPING)
```

But turns out this Python 3 only.

Reply:
we could keep a modified copy a an attribute to the module.

Reply:
What are we solving by keeping a copy?

Reply:
we stop changing the dict that the user of the module provides us. Also he cannot accidentally change our copy midway. Otoh it kind of binds the module instance to a specific entity class.
In the end it comes back to my question whether this imposes a real world problem. I am ok with saying: no.

Reply:
I say no. :)